src/lib.rs

```diff
@@ -25,18 +25,23 @@ use std::fmt::Debug;
 /// Navigator's default crystal clock for PWM, with a value of 24.5760 MHz
 const NAVIGATOR_PWM_XTAL_CLOCK_FREQ: f32 = 24_576_000.0;
 
+/// The `AxisData` struct represents a set of three floating-point values for the x, y, and z axes.
 #[derive(Debug)]
 pub struct AxisData {
     pub x: f32,
     pub y: f32,
     pub z: f32,
 }
 
+/// The `ADCData` struct represents data from ADC with it's four channels,
+/// each storing a 16-bit signed integer value.
 #[derive(Debug)]
 pub struct ADCData {
     pub channel: [i16; 4],
 }
 
+/// The `SensorData` struct represents data collected from various sensors, including ADC, temperature,
+/// pressure, accelerometer, magnetometer, and gyro.
```

Comment:
This is too generic, it should be something line:
```suggestion
/// Encapsulates the value of all sensors on the board. 
```